### PR TITLE
initial pundit setup completed

### PR DESCRIPTION
**Pull request by Glod33**
So the initial setup is done, once a models are finished I will add pundit to it in a PATCH issue/PR.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,21 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
```
Comment (wJoenn) [inline app/controllers/application_controller.rb:2]: I didn't realize Pundit required Devise before being installed but this is a Devise method and our application does not include Devise yet.
That means we'll have to wait until https://github.com/wJoenn/artify/issues/4 is completed before merging this PR, sorry about that.

File: app/policies/application_policy.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    def resolve
+      raise NotImplementedError, "You must define #resolve in #{self.class}"
+    end
+
+    private
+
+    attr_reader :user, :scope
+  end
+end
```
Comment (wJoenn) [inline app/policies/application_policy.rb:53]: This you edit this file on any way or is this automatically generated ?

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,2 +1,21 @@
...
+  include Pundit::Authorization
```
Comment (wJoenn) [inline app/controllers/application_controller.rb:3]: I didn't realize Pundit required Devise before being installed but `authenticate_user` is a Devise method and our application does not include Devise yet.
That means we'll have to wait until https://github.com/wJoenn/artify/issues/4 is completed before merging this PR, sorry about that.

**Review by wJoenn — CHANGES_REQUESTED** [review]

File: app/policies/application_policy.rb (hunk 2 above)
Comment (Glod33) [reply app/policies/application_policy.rb:53]: automatically generated

File: app/controllers/application_controller.rb (hunk 3 above)
Comment (Glod33) [reply app/controllers/application_controller.rb:3]: cool! no problem:)

File: app/policies/application_policy.rb (hunk 2 above)
Comment (MarianaCarlota) [reply app/policies/application_policy.rb:53]: I think you ca allready update the show to true since everyone can see the show page

Comment (MarianaCarlota) [reply app/policies/application_policy.rb:53]: yeah, you are right. sorry

Comment (wJoenn) [reply app/policies/application_policy.rb:53]: There's really no need for an apology Mariana, no worries 🙂

**Review by wJoenn — APPROVED** [review]

**Pull request merged**